lib/MusicBrainz/Server/Data/Release.pm

```diff
@@ -138,8 +138,12 @@ sub _where_filter
             push @params, $filter->{artist_credit_id};
         }
         if (exists $filter->{label_id}) {
-            push @query, 'EXISTS (SELECT 1 FROM release_label rl WHERE rl.release = release.id AND rl.label = ?)';
-            push @params, $filter->{label_id};
+            if ($filter->{label_id} eq '-1') {
+                push @query, '(NOT EXISTS (SELECT 1 FROM release_label rl WHERE rl.release = release.id) OR EXISTS (SELECT 1 FROM release_label rl WHERE rl.release = release.id AND rl.label IS NULL))';
```

Comment:
I guess it's intended to return releases where any of the labels are null, not where all labels are null?

Reply:
Yes, I thought it could be useful to also show ones where some labels are empty but has a catno, since there's no way to find them otherwise and it should be trivial to separate them by checking the list.